### PR TITLE
Refactor file locking code for testing

### PR DESCRIPTION
**Pull request by vmrob**

### CODE REVIEW

**Comment by vmrob** [comment]
I take that back... The `FileLock` context manager isn't used anywhere. Let me remove that and if it's ever to be used, it should be easy to add it back.

File: needy/filesystem.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    lock_acquired = False
+    while True:
+        try:
+            fcntl.flock(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
+            lock_acquired = True
+        except IOError:
+            pass
+        if lock_acquired or datetime.now() > start + timedelta(seconds=timeout):
+            break
+        time.sleep(1)
```
Comment (ccbrown) [inline needy/filesystem.py]: i really don't like this way of implementing a timeout. if the timeout is necessary, we should probably do it some other way...

http://stackoverflow.com/questions/5255220/fcntl-flock-how-to-implement-a-timeout

Comment (vmrob) [inline needy/filesystem.py]: I can work around it, but it's worth mentioning that the suggested method is somewhat global to the process even if it is better.

Comment (ccbrown) [inline needy/filesystem.py]: I don't think we have any plans or motivation to multithread Needy itself, so I'd definitely prefer it.

Comment (vmrob) [inline needy/filesystem.py]: That's about what I thought. Definitely better to just allow multiple single-threaded Needy instances.

**Comment by vmrob** [comment]
Much cleaner.

File: needy/local_configuration.py [hunk 2]
```diff
@@ -21,14 +23,12 @@ def __enter__(self):
...
-        try:
-            fcntl.flock(self.__fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
-        except IOError:
+        self.__fd = lock_file(self.__path, timeout=0)
+        if self.__fd is None:
             if not self.__blocking:
                 return None
             print('Waiting for other needy instances to terminate...')
-            fcntl.flock(self.__fd, fcntl.LOCK_EX)
+            self.__fd = lock_file(self.__path, timeout=None)
```
Comment (ccbrown) [inline needy/local_configuration.py]: you could leave off the `timeout=None` here

**Comment by ccbrown** [comment]
Looks good. I'll merge it once Travis approves.

**Comment by vmrob** [comment]
Removed the `timeout=None`.

**Pull request merged**